[{"count": 0, "tags": [], "bug_id": 47570, "is_private": false, "text": "Use Mac Excel 2008 to create a workbook with a formula.\n\nRun a POI-based program that reads in the workbook, deletes the sheet containing the formula, creates a new sheet, and writes the result out as a new workbook.\n\nMac Excel 2008 opens the new workbook fine, as does NeoOffice and the Finder's built-in file viewer. However, Excel 2007 on Windows complains of unreadable content and offers to remove it. It then reports removing the following:\n\n<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\" ?> \n- <recoveryLog xmlns=\"http://schemas.openxmlformats.org/spreadsheetml/2006/main\">\n  <logFileName>error005320_01.xml</logFileName> \n  <summary>Errors were detected in file 'D:\\DynSQL\\FRx\\IO_Data\\workbook.xlsx'</summary> \n- <removedRecords summary=\"Following is a list of removed records:\">\n  <removedRecord>Removed Records: Formula from /xl/calcChain.xml part (Calculation properties)</removedRecord> \n  </removedRecords>\n  </recoveryLog>\n\nafter which it successfully opens the file.\n\nIf you modify the POI-based program to delete all the content in the sheet to be deleted before deleting the sheet, the problem still occurs. However, if you also set all the cell formulas to null, then the problem disappears.", "id": 129102, "time": "2009-07-23T18:06:54Z", "creator": "prescindor@gmail.com", "creation_time": "2009-07-23T18:06:54Z", "attachment_id": null}, {"count": 1, "tags": [], "creator": "yegor@dinom.ru", "attachment_id": null, "text": "I can't reproduce it with files created in Excel 2007.\n\nCould you please attach the following data:\n\n - source .xlsx file created by Mac Excel 2008\n - Java code that modifies it\n - output file that is unreadable by Excel 2007\n\nYegor", "id": 129574, "time": "2009-08-08T04:14:54Z", "bug_id": 47570, "creation_time": "2009-08-08T04:14:54Z", "is_private": false}, {"count": 2, "tags": [], "creator": "dominik.stadler@gmx.at", "attachment_id": null, "text": "no response in a long time => resolving for now, please reopen with more information if this is still an issue", "id": 169112, "time": "2013-08-05T09:43:23Z", "bug_id": 47570, "creation_time": "2013-08-05T09:43:23Z", "is_private": false}, {"count": 3, "tags": [], "bug_id": 47570, "is_private": false, "text": "(In reply to Dominik Stadler from comment #2)\n> no response in a long time => resolving for now, please reopen with more\n> information if this is still an issue\n\nI can confirm this issue is still happening and I'm using 3.10-beta2.\n\nI've attached a very simple test to verify this issue using Groovy.\n\nBasically, test.xlsx is the original spreadsheet with no problems and contains two very simple formulas in cells A1 and A2.  testOut.xlsx is the output after running sheet.createRow(0) on test.xlsx.\n\nWhat's expected in testOut.xlsx is the first becomes blank whilst second row still has the formula from cell A2.\n\nWhat actually happens is that an error from Excel saying:\n\"Excel could not open testOut.xlsx because some content is unreadable.  Do you want to open and repair this workbook?\n\nSelecting open and repair fixes the problem and the \"Review Log File\" from Excel contains this:\n<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>\n<recoveryLog xmlns=\"http://schemas.openxmlformats.org/spreadsheetml/2006/main\"><logFileName>Repair Result to testOut 03366.xml</logFileName><summary>Errors were detected in file 'Root:Development:tmp:testOut.xlsx'</summary><removedRecords summary=\"Following is a list of removed records:\"><removedRecord>Removed Records: Formula from /xl/calcChain.xml (Calculation properties)</removedRecord></removedRecords></recoveryLog>\n\nI also found related posts on previous POI mailing list:\nhttp://mail-archives.apache.org/mod_mbox/poi-user/201009.mbox/%3C4C988E0F.3030802@science-computing.de%3E\n\nA reply to this suggested \"workbook.onDeleteFormulaCell wasn't being called\"... not sure if it's related.\nhttp://mail-archives.apache.org/mod_mbox/poi-user/201009.mbox/%3Calpine.DEB.1.10.1009211216380.337@urchin.earth.li%3E\n\nI also verified that instead of sheet.createRow(0), I do, sheet.removeRow() works and does not produce the error.", "id": 170824, "time": "2013-10-25T06:46:14Z", "creator": "tommy.li@firefire.co", "creation_time": "2013-10-25T06:46:14Z", "attachment_id": null}, {"count": 4, "tags": [], "bug_id": 47570, "is_private": false, "text": "Created attachment 30963\ntest script in groovy", "id": 170825, "time": "2013-10-25T06:49:11Z", "creator": "tommy.li@firefire.co", "creation_time": "2013-10-25T06:49:11Z", "attachment_id": 30963}, {"count": 5, "tags": [], "bug_id": 47570, "text": "Created attachment 30964\ntest input to script", "id": 170826, "time": "2013-10-25T06:49:27Z", "creator": "tommy.li@firefire.co", "creation_time": "2013-10-25T06:49:27Z", "is_private": false, "attachment_id": 30964}, {"count": 6, "tags": [], "bug_id": 47570, "text": "Created attachment 30965\ntest output to script", "id": 170827, "attachment_id": 30965, "creator": "tommy.li@firefire.co", "creation_time": "2013-10-25T06:49:44Z", "time": "2013-10-25T06:49:44Z", "is_private": false}, {"count": 7, "tags": [], "bug_id": 47570, "is_private": false, "text": "Hello,\n\nI'm having the same problem with all the latest version I tested.\nDo you have any feedback on this issue?\n\nThanks in advance", "id": 175875, "time": "2014-06-18T20:06:21Z", "creator": "ponteandres@hotmail.com", "creation_time": "2014-06-18T20:06:21Z", "attachment_id": null}, {"count": 8, "tags": [], "creator": "yegor@dinom.ru", "attachment_id": null, "text": "This is no longer reproducible with 4.0.0-SNAPSHOT.\nI ran the Groovy snippet and it produced a valid file. Excel 2013 opens it without issues.\n\nYegor", "id": 202885, "time": "2017-12-21T14:04:23Z", "bug_id": 47570, "creation_time": "2017-12-21T14:04:23Z", "is_private": false}]